pandas/core/reshape/merge.py

```diff
@@ -1200,9 +1219,32 @@ def _maybe_coerce_merge_keys(self):
                 typ = rk.categories.dtype if rk_is_cat else object
                 self.right = self.right.assign(**{name: self.right[name].astype(typ)})
 
+    def _create_cross_configuration(
+        self, _left, _right
+    ) -> Tuple["DataFrame", "DataFrame", str, str]:
```

Comment:
can you give a doc-sstring here on what is going on

Reply:
and also parameters / returns

Reply:
Good point, added clarification